toolchain/sem_ir/typed_insts.h

```diff
@@ -833,6 +833,10 @@ struct Param {
 
   TypeId type_id;
   RuntimeParamIndex runtime_index;
+
+  // A name to associate with this Param in pretty-printed IR. This is not
+  // necessarily unique, or even valid, and has no semantic significance.
+  NameId pretty_name;
```

Comment:
Have you considered including the `InstId` of the pattern here instead of this name? It seems like we can in general reconstruct this from the pattern, and storing the pattern would likely be more generally useful.

Reply:
See my response to your comment on `handle_function.cpp`, where I call this option 3.